learntools/ml-insights/nbs/ch3.ipynb

```diff
@@ -0,0 +1,836 @@
M5p1ew78KqRazPMJs37TANRdwrb8HrydQhX4SfxGs/6zcGFjqWDwk1ep\nWF/LAXQQ/RZDNOwz2z5lAD4U34eK6VeLdXIzePGq+fZbDWAteLFtovl6a+LyXNB4q7nrzbIb+FXT\n72JbZ4IXwRoh7sSoZ5m/FLH/2sRltgOvoskBPz4+BbAborGaGOZm8OqtKvDGe/MBeDZzvfQX06gE\nbzuyAryUoQT8h2eEWHfBZuNc0A285Ox78Cv4PPD9Ph3NaMwoptMB/DgrBb+inY7/NmbcUs94jW5L\nMUwe+L45DxeeZ/zA99ci8P1yI8SdSqL/ZW3MKL5PFtu9BrwN0ZsA7Opbvyrr3AM8OSgBv2PiPTGP\n/6kcnxecW8AbnJqO7WQRzylceIy8AN6mqAI8+fsFZo2x0cg5kv6a9me6jYWQFkfcez1fUZR3bSCW\n7eAnvYetHQsAMMZ8wIt9Nyj84V5gjPmDJwzDFEWxbAhIrITx5x7MVxSl1bQZEyVxowFsVhSlXHSz\nB/+xnqEoyrKGxie2haoeCGmdDOBXT7KoNusN4DvwK7L6WuoTcjlVg1effCmq4iLF91rwBqmkBaFE\ngZArjPGnF5Y18hd+OeepKEoh+PMHdOBVZjvAi7dvUhTlSrWUJzZItDNobP8bcTnnqfCi6tvAq/AO\niL+OAHQKb4hIWhCqeiDkCmP8UcNqdxuYnKUfcHIlMMY80fgdHucURVG7DZO0cZQoEEIIIUQVVT0Q\nQgghRBUlCoQQQghRRYkCIYQQQlRRokAIIYQQVZQoEEIIIURVq3kSWFPIshwG/ijSRyRJmm/W/SUA\nPSVJmnaZ5nMGwN2SJB28HNNrZF7u4A8w8QDwliRJqy9xeuMA3ChJUn1vXLMKWZbfBnBKkqSlFt19\nAORKknRZ3lAny/J8AD9JkrSlgWE6AZgjSdJdlzivZQA+kiQp4WLjEftzgiRJrpcSS2Ou5HzMt60s\ny28BOCJJ0rpmTqMM/Pg9c4mxRAPYIElSgFm3geCPU24H/vjoKZIkNfnNl2IafwD4RZKkxZcSXxPn\n9QaAx8AfW/wF+GPFiwHcqbZ+ZFneDuArSZJ+ucyx9AfwmiRJd8uyfC2AhyRJeryZ0/gKQJ4kSbMu\nMRZHAH8A+M60nLIsu4A/Urwv+EXzq5IkrZVlWQv+bpAHJUnKuZT5thZtKlEQjADmyLL8tyRJreHF\nIH0A+EmS1OVyTEySpN/w7wugbIIkSW9dpfk05fHLHQFENDpUA2RZvhdAcUNJQjPiadEstu1IXLm3\nIaqSZdkO/CVNr4EnBKbuDuDvDpggSdJuWZafALAAwJirHWMzPARgkiRJu0Titc1a+5G4ULpbfO0B\n/q6Vq06W5UHgT4WMBH86qcksAGWSJEXJshwKYK8sywclSUqXZXm2GOfu/0ywDWqLiUIlgE8ArJBl\neZAkSRe8T16W5cXgV09zLL+LkoIfAdwK/gATCfxVrNeAP5p0nCRJGWJSM8QViiOATyRJWiimNxb8\nBS4O4C8qeUmSpD2yLM8CfyNeAIA4SZKmWMR1u5ifFvwlKy+AXyksBBAky3IsgEGSJFWajbMdZlcK\n5t9lWa4C8CGAmwAEAvhCkqTPZVmeBl4acpssyz3AM+524CfwMPCT6RmYXWFaXnHKsvwQgCfBs/R8\nAE9JknTcckPIsvwa+ImtFPyNnLdLkhQm1rkXgM7gVwF+ZtvgTvCXy1SAP+2tXrIs14G/RfIGEf/r\nkiStEf3eBH8jZh34y6+ekiQpy7R+wN+6qAfwJ/j7ErzAXwb2i1gfQbIsbwTfD+bi39fvJgOYLklS\nmSgNOChJ0rf1hQfgHhHLNBGLBkAQ+IuGpkqSlGGxvW4Df6OfBkA5gMfBt79peaNEvC+AvyWz3u3T\n0PzU1qXFerUHf/mTDvwx0fsAPC9JUqk4PhaLfqEAVkqS9IoYr6FtnQB+XPYH8LEsywYA46F+HF4v\n1rsCvg+cr0Jt4PgKFOtnTD3L2g9Ab/AfBfPHC18LoESSpN3i+wIAn8uy7C1JUn4D6ygQwBLw4+os\n+IukTP2qwa9Wo8FfcuQM4GPwl1vVAJgpSdKGRvaLYADfgB+PDMASSZI+lmV5JfiP8QJZlt8HPwa1\nsiw7S5I0WS1ei9j/c56RJGm/OD+FgZ+fOgLIBXCfiGcA+I+qA/iLrzqC74cAP55uAX/zZHtZlheJ\ndfOVJEk9xTxHmL6LEtL5Yv1kgh+ju8RwQWJ6oQDswUvb3hf9GjrengHfJ1626H4HgEkAIElSqizL\nmwDcC+BTSZL+lmX5W1mWoyVJOoI2rq22UXgP/GT7/kWM6yRJUjSAF8HfRveF+J4G/hY4k0pJkvqB\n/xB/KMtyD1mWu4p5jpEkqS+ARwGskWXZdBXTEUC/epKESPBXrN4lSVJvAG+Bn2wyATwM4LQkSX3M\nk4QmcAQv0hsCfoL8UJZlJ4thVgCYL+Y5D/wVzg2SZXk4gKkArhfLOBvAmnqGGwW+vq4FT7TcLAZx\nkSSphyRJr5qN4weeGN0lSdI14CdhNVoABWK4ewEslGXZV5bl6eAnrmvFciWA/7hZCgewUZKkAeBv\nKpwtSZIB/67vUeCJ3QgAvcV8ksF/cCBJ0sP1nbRkWe4JwNmiNGEIgBmSJHUHf1vjlxbj+IG/hW+a\niPlj8CTPfJq/A3hYkqRfG1gnTZpfI2aC/wBGiz+NiMfEVZKk6wEMBvC0LMudmrCtIUnSPPAE7eWG\nlkFc5a8C8KLYv7aB/9iioeNLkqQMcYz8JyGSJGm/JEnT8e9r2E1CzLuJi4pc8B/uhswDsFeSpB7g\nP1KRZv0cAPwuSVIEeDXoLwCeFdt1KoAfRPUWoL6dloOXFPQSw0yRZXmCJEn3gVePTJYkaQn4OWNl\nM5KEes8z4scbAK4HcI8kSZHgb+F8TJTGrAbwphjnS/BSzvMkSUoT09op
```

Comment:
This is a really cool question. I love a challenging brainteaser! But I don't think the current solution(?) code works. (Not sure if it's intended to be a solution, or just a starting point)

Reply:
Specifically I don't think it works because it seems that the PDP is not flat.